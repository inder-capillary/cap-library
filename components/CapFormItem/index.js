import React from "react";
import PropTypes from 'prop-types';
import { Form } from "antd";
import "./_capFormItem.scss";
const classNames = require('classnames');
const FormItem = Form.Item;
export default class CapFormItem extends React.Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {className, children, ...rest} = this.props;
    return (
      <FormItem {...rest} className={classNames("cap-form-item-v2", className)}>
        {React.Children.toArray(children)}
      </FormItem>
    );
  }
}

CapFormItem.propTypes = {
  children: PropTypes.any,
  className: PropTypes.string,
};
