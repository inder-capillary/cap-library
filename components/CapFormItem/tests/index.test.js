import renderer from 'react-test-renderer';
import React from 'react'; // eslint-disable-line no-unused-vars
// import {mount} from 'enzyme';

import CapFormItem from '../index';

describe('<CapFormItem />', () => {
  it('CapFormItem renders correctly', () => {
    const tree = renderer.create(<CapFormItem />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
