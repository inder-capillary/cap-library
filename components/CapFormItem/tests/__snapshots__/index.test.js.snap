// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CapFormItem /> CapFormItem renders correctly 1`] = `
<div
  className="ant-row ant-form-item cap-form-item-v2"
>
  <div
    className="ant-form-item-control-wrapper"
  >
    <div
      className="ant-form-item-control"
    >
      <span
        className="ant-form-item-children"
      />
    </div>
  </div>
</div>
`;
